src/components/transactions/Supply/SupplyModalContent.tsx

```diff
@@ -41,209 +41,214 @@ export enum ErrorType {
-  const totalCollateralMarketReferenceCurrencyAfter = user
-    ? valueToBigNumber(user.totalCollateralMarketReferenceCurrency).plus(amountIntEth)
-    : '-1';
-
-  const liquidationThresholdAfter = user
-    ? valueToBigNumber(user.totalCollateralMarketReferenceCurrency)
-        .multipliedBy(user.currentLiquidationThreshold)
-        .plus(amountIntEth.multipliedBy(poolReserve.formattedReserveLiquidationThreshold))
-        .dividedBy(totalCollateralMarketReferenceCurrencyAfter)
-    : '-1';
-
-  let healthFactorAfterDeposit = user ? valueToBigNumber(user.healthFactor) : '-1';
-
-  if (
-    user &&
-    ((!user.isInIsolationMode && !poolReserve.isIsolated) ||
-      (user.isInIsolationMode &&
-        user.isolatedReserve?.underlyingAsset === poolReserve.underlyingAsset))
-  ) {
-    healthFactorAfterDeposit = calculateHealthFactorFromBalancesBigUnits({
-      collateralBalanceMarketReferenceCurrency: totalCollateralMarketReferenceCurrencyAfter,
-      borrowBalanceMarketReferenceCurrency: valueToBigNumber(
-        user.totalBorrowsMarketReferenceCurrency
-      ),
-      currentLiquidationThreshold: liquidationThresholdAfter,
-    });
-  }
-
-  // ************** Warnings **********
-  // isolation warning
-  const hasDifferentCollateral = user.userReservesData.find(
-    (reserve) => reserve.usageAsCollateralEnabledOnUser && reserve.reserve.id !== poolReserve.id
-  );
-  const showIsolationWarning: boolean =
-    !user.isInIsolationMode &&
-    poolReserve.isIsolated &&
-    !hasDifferentCollateral &&
-    (userReserve && userReserve.underlyingBalance !== '0'
-      ? userReserve.usageAsCollateralEnabledOnUser
-      : true);
-
-  // TODO: check if calc is correct to see if cap reached
-  const capReached =
-    poolReserve.supplyCap !== '0' &&
-    valueToBigNumber(amount).gt(
-      new BigNumber(poolReserve.supplyCap).minus(poolReserve.totalLiquidity)
+    poolReserve,
+    userReserve,
+    isWrongNetwork,
+    nativeBalance,
+    tokenBalance,
+  }: ModalWrapperProps) => {
+    const { marketReferencePriceInUsd, user } = useAppDataContext();
+    const { currentMarketData, currentNetworkConfig } = useProtocolDataContext();
+    const { mainTxState: supplyTxState, gasLimit, txError } = useModalContext();
+    const { supplyCap: supplyCapUsage, debtCeiling: debtCeilingUsage } = useAssetCaps();
+    const {
+      poolComputed: { minRemainingBaseTokenBalance },
+    } = useRootStore();
+
+    // states
+    const [amount, setAmount] = useState('');
+    const supplyUnWrapped = underlyingAsset.toLowerCase() === API_ETH_MOCK_ADDRESS.toLowerCase();
+
+    const walletBalance = supplyUnWrapped ? nativeBalance : tokenBalance;
+
+    const supplyApy = poolReserve.supplyAPY;
+    const { supplyCap, totalLiquidity, isFrozen, decimals, debtCeiling, isolationModeTotalDebt } =
+      poolReserve;
+
+    // Calculate max amount to supply
+    const maxAmountToSupply = useMemo(
```

Comment:
Kept this one in since the higher level `useMemo` re-renders on changes to `poolReserve` whereas this destructures to only update on sub-fields of the reserve that are used

It's probably overkill, but it does serve a small purpose